share/eds/eds.go

```diff
@@ -45,28 +46,35 @@ type writingSession struct {
 // Order: [ Carv1Header | Q1 |  Q2 | Q3 | Q4 | inner nodes ]
 // For more information about the header: https://ipld.io/specs/transport/car/carv1/#header
 func WriteEDS(ctx context.Context, eds *rsmt2d.ExtendedDataSquare, w io.Writer) error {
+	ctx, span := tracer.Start(ctx, "write-eds")
+	defer span.End()
+
 	// 1. Reimport EDS. This is needed to traverse the NMT tree and cache the inner nodes (proofs)
 	writer, err := initializeWriter(ctx, eds, w)
 	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
 		return fmt.Errorf("share: creating eds writer: %w", err)
```

Comment:
* Writing the error we compose here should be more readable.
* Try defining errors at the top of the methods and then do one defer that sets the status if the error is not nil. 
* Also, there is a `RecordError` API that specifically takes errors, and let's use it instead of passing the error string to the status. Note that `RecordError` still requires setting the status.